### DISCUSSION TITLE
V3 layout @section() and @yeild  doesn't work when using full page component

### DISCUSSION THREAD

**Discussion by onemoreahmad** [livewire/livewire#5914] [7. Bugs (archived)]
**Livewire : ^3.0@beta**

From the documentation :
 
<img width="776" alt="Screen Shot 2023-07-24 at 3 24 19 AM" src="https://github.com/livewire/livewire/assets/1885785/0b9df671-f2c7-471b-bd0a-10eabb5e61a7">

If this layout uses [Template Inheritance](https://laravel.com/docs/10.x/blade#layouts-using-template-inheritance), then I should be able to add `@yield()` or `@section()` like so 

```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
 
        <title>{{ $title ?? 'Page Title' }}</title>
    </head>
    <body>
        @yield('heading')
        {{ $slot }}
    </body>
</html>
```
and inside my full page component I can use it like normally do:

```
<div>
    @section('heading')
        <p>This is component heading.</p>
    @endsection

    component content is here. 
</div>
```

But if Livewire layouts a blade components, then I should be able to use [slots](https://laravel.com/docs/10.x/blade#slots) 

```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
 
        <title>{{ $title ?? 'Page Title' }}</title>
    </head>
    <body>
        {{ $heading ?? ''}}
        {{ $slot }}
    </body>
</html>
```

Then in my component : 
```
<div>
    <x-slot:heading>
        <p>This is component heading.</p>
    </x-slot>
 
    component content is here. 
</div>
```

**But none of the above works.**

**Comment by WyattCast44** [comment]
As a workaround using the syntax below still works 👍

```php
 public function render()
{
    return view('auth.login')->extends('layouts.auth')->section('content');
}
```

**Reply by onemoreahmad** [reply]
Thanks @WyattCast44 , unfortunately this is also doesn't work. 
Here, when you add `->section('anything')` you're telling Livewire that you want the content of the component to be inside section `anything` but you can't add new section for example `@section('heading')` to the layout and use it in the component as far as I understand.

**Comment by joshhanley** [comment]
@onemoreahmad thanks for reporting! This problem has already been reported in #5846 so we will continue the discussion there.
